chain/rosetta-rpc/src/adapters/transactions.rs

```diff
@@ -268,7 +322,14 @@ fn convert_account_update_to_operations(
                 amount: Some(-crate::models::Amount::from_yoctonear(deposit)),
                 type_: crate::models::OperationType::Transfer,
                 status: Some(crate::models::OperationStatusKind::Success),
-                metadata: None,
+                metadata: if let Some(predecessor_id) = predecessor_id {
+                    Some(OperationMetadata {
+                        predecessor_id: Some(predecessor_id.clone()),
+                        ..Default::default()
+                    })
+                } else {
+                    None
+                },
```

Comment:
This is being repeated over and over.  Would be best to move it into its own function or a `from_predecessor` method on `OperationMetadata`.